source/s3_util.c

```diff
@@ -107,3 +108,99 @@ struct aws_string *get_top_level_xml_tag_value(
 
     return xml_user_data.result;
 }
+
+struct aws_cached_signing_config_aws *aws_cached_signing_config_new(
+    struct aws_allocator *allocator,
+    const struct aws_signing_config_aws *signing_config) {
+    AWS_PRECONDITION(allocator);
+    AWS_PRECONDITION(signing_config);
+
+    struct aws_cached_signing_config_aws *cached_signing_config =
+        aws_mem_calloc(allocator, 1, sizeof(struct aws_cached_signing_config_aws));
+
+    cached_signing_config->allocator = allocator;
+
```

Comment:
I was thinking about that, but I was worried about somebody adding pointers upstream and forgetting to update the client, so I opted for explicit copies. :-/  If someone does forget, then those settings won't get copied, but at least it'll pick up defaults instead of copying a pointer.

Reply:
saved by calloc I guess